Source/Core/InputCommon/ControllerInterface/OSX/OSXJoystick.mm

```diff
@@ -2,75 +2,71 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include "InputCommon/ControllerInterface/OSX/OSXJoystick.h"
+
 #include <algorithm>
 #include <sstream>
 
 #include <Foundation/Foundation.h>
 #include <IOKit/hid/IOHIDLib.h>
 
+#include "Common/Logging/Log.h"
 #include "Common/StringUtil.h"
-#include "InputCommon/ControllerInterface/OSX/OSXJoystick.h"
 
 namespace ciface
 {
 namespace OSX
 {
-Joystick::Joystick(IOHIDDeviceRef device, std::string name)
-    : m_device(device), m_device_name(name), m_ff_device(nullptr)
+void Joystick::AddElements(CFArrayRef elements)
 {
-  // Buttons
-  NSDictionary* buttonDict = @{
-    @kIOHIDElementTypeKey : @(kIOHIDElementTypeInput_Button),
-    @kIOHIDElementUsagePageKey : @(kHIDPage_Button)
-  };
+  for (int i = 0; i < CFArrayGetCount(elements); i++)
+  {
+    IOHIDElementRef e = (IOHIDElementRef)CFArrayGetValueAtIndex(elements, i);
 
-  CFArrayRef buttons =
-      IOHIDDeviceCopyMatchingElements(m_device, (CFDictionaryRef)buttonDict, kIOHIDOptionsTypeNone);
+    uint32_t usage = IOHIDElementGetUsage(e);
```

Comment:
Looks like these could be `const` and you could declare `usage`/`usage_page` down lower where they are actually used.